src/gm-apclientpp.cpp

```diff
@@ -684,3 +719,192 @@ double apclient_death_link(const char* cause)
         return GM_FALSE;
     return GM_BOOL(apclient && apclient->Bounce(data_j, {}, {}, { "DeathLink" }));
 }
+
+inline int key_to_int(const char* key) {
+    int i = 0;
+    while (*key) {
+        i *= 10;
+        i += *key++ - '0';
+    }
+    return i;
+}
```

Comment:
This should have return type `json::size_type` (which an alias for `size_t`) and should be static (i.e. not visible after compilation). To get a size_t from a string, you can use strtoull with base=10 and then cast on return to make sure to get rid of platform differences.
```suggestion
static inline json::size_type key_to_int(const char* key)
{
    return (json::size_type) strotull(key, nullptr, 10)
}
```
This does not validate that the key was valid, but I don't think that's a concern.